api/handler/multipart_upload.go

```diff
@@ -0,0 +1,536 @@
+		NextpartNumberMarker int           `xml:"NextpartNumberMarker,omitempty"`
+		Owner                Owner         `xml:"Owner"`
+		Parts                []*layer.Part `xml:"Part"`
+		PartNumberMarker     int           `xml:"PartNumberMarker,omitempty"`
+		StorageClass         string        `xml:"StorageClass,omitempty"`
+		UploadId             string        `xml:"UploadId"`
+	}
+
+	MultipartUpload struct {
+		Initiated    string    `xml:"Initiated"`
+		Initiator    Initiator `xml:"Initiator"`
+		Key          string    `xml:"Key"`
+		Owner        Owner     `xml:"Owner"`
+		StorageClass string    `xml:"StorageClass,omitempty"`
+		UploadID     string    `xml:"UploadId"`
+	}
+
+	Initiator struct {
+		ID          string `xml:"ID"`
+		DisplayName string `xml:"DisplayName"`
+	}
+
+	CompleteMultipartUpload struct {
+		XMLName xml.Name               `xml:"http://s3.amazonaws.com/doc/2006-03-01/ CompleteMultipartUpload"`
+		Parts   []*layer.CompletedPart `xml:"Part"`
+	}
+)
+
+const (
+	uploadIdHeaderName   = "uploadId"
+	partNumberHeaderName = "partNumber"
+	maxSizeUploadsList   = 1000
+	maxSizePartsList     = 1000
+	uploadMaxPartNumber  = 10000
+)
+
+func (h *handler) CreateMultipartUploadHandler(w http.ResponseWriter, r *http.Request) {
+	reqInfo := api.GetReqInfo(r.Context())
+
+	bktInfo, err := h.obj.GetBucketInfo(r.Context(), reqInfo.BucketName)
+	if err != nil {
+		h.logAndSendError(w, "could not get bucket info", reqInfo, err)
+		return
+	}
+
+	if err := checkOwner(bktInfo, r.Header.Get(api.AmzExpectedBucketOwner)); err != nil {
+		h.logAndSendError(w, "expected owner doesn't match", reqInfo, err)
+		return
+	}
+
+	metadata := parseMetadata(r)
+	uploadID := uuid.New()
+
+	additional := []zap.Field{zap.String("uploadID", uploadID.String()), zap.String("Key", reqInfo.ObjectName)}
+	uploadInfo := &layer.UploadInfoParams{
+		UploadID: uploadID.String(),
+		Bkt:      bktInfo,
+		Key:      reqInfo.ObjectName,
+	}
+
+	p := &layer.UploadPartParams{
+		Info:       uploadInfo,
+		PartNumber: 0,
+		Header:     metadata,
+	}
+
+	info, err := h.obj.UploadPart(r.Context(), p)
+	if err != nil {
+		h.logAndSendError(w, "could not upload a part", reqInfo, err, additional...)
+	}
+
+	var newEaclTable *eacl.Table
+	if containsACLHeaders(r) {
+		if newEaclTable, err = h.getNewEAclTable(r, info); err != nil {
+			h.logAndSendError(w, "could not get new eacl table", reqInfo, err, additional...)
+			return
+		}
+	}
+	if newEaclTable != nil {
+		p := &layer.PutBucketACLParams{
+			Name: reqInfo.BucketName,
+			EACL: newEaclTable,
+		}
+
+		if err = h.obj.PutBucketACL(r.Context(), p); err != nil {
+			h.logAndSendError(w, "could not put bucket acl", reqInfo, err, additional...)
+			if err = h.obj.AbortMultipartUpload(r.Context(), uploadInfo); err != nil {
+				h.log.Error("could not delete multipart upload init file", additional...)
+			}
+			return
+		}
+	}
+
+	tagSet, err := parseTaggingHeader(r.Header)
+	if tagSet != nil {
```

Comment:
I would expect
```suggestion
	if err == nil {
```

or even 

```go
if err != nil {
    h.logAndSendError(...)
    h.obj.AbortMultipardUpload(...)
}
h.obj.PutObjectTagging(...)
```